husky_directory/services/translator.py

```diff
@@ -126,10 +38,41 @@ def _resolve_phones(
 
         return model
 
+    @staticmethod
+    def _translate_student_attributes(
+        student: StudentPersonAffiliation,
+        result_in_progress: Person,
+    ) -> NoReturn:
+        result_in_progress.email = student.directory_listing.email
+        result_in_progress.departments.extend(
+            UWDepartmentRole(
+                title=student.directory_listing.class_level, department=dept
+            )
+            for dept in student.directory_listing.departments
+            if dept  # Ignore data with holes in it.
+        )
+
+    @staticmethod
+    def _translate_employee_attributes(
+        employee: EmployeePersonAffiliation, result_in_progress: Person
+    ) -> NoReturn:
+        # Email will usually be the same, but just in case, we'll prefer the
+        # employee email address and not overwrite it with the student's if
+        # it's already set.
+        if employee.directory_listing.emails:
+            result_in_progress.email = employee.directory_listing.emails[0]
+
+        result_in_progress.box_number = employee.mail_stop
+        result_in_progress.departments.extend(
+            UWDepartmentRole.from_orm(position)
+            for position in employee.directory_listing.positions
+            # Sometimes the data we get has holes in it;
+            # we ignore holey data.
+            if position.department and position.title
+        )
+
```

Comment:
This code didn't really change, just got moved into separate methods from a monster-block before.